src/main/java/frc/team2412/robot/subsystems/IntakeOnOffSubsystem.java

```diff
@@ -32,82 +32,126 @@ public IntakeOnOffSubsystem(CANSparkMax frontMotor, CANSparkMax backMotor) {
-		RobotState.m_intakeDirection = IntakeDirection.BACK;
-	}
-
-	public void frontIntakeIn() {
-		m_intakeFrontMotor.set(IntakeConstants.MAX_INTAKE_SPEED);
-		m_lastMotor = IntakeLastMotor.FRONT;
-		RobotState.m_intakeDirection = IntakeDirection.FRONT;
-	}
-
-	public void frontIntakeOut() {
-		m_intakeFrontMotor.set(-IntakeConstants.MAX_INTAKE_SPEED);
-		m_lastMotor = IntakeLastMotor.FRONT;
-		RobotState.m_intakeDirection = IntakeDirection.FRONT;
-	}
-
-	public void frontIntakeInBackIntakeOff() {
-		m_intakeFrontMotor.set(IntakeConstants.MAX_INTAKE_SPEED);
-		m_intakeBackMotor.set(0);
-		m_lastMotor = IntakeLastMotor.FRONT;
-	}
-
-	public void frontIntakeOutBackIntakeOff() {
-		m_intakeFrontMotor.set(IntakeConstants.MAX_INTAKE_SPEED);
-		m_intakeBackMotor.set(0);
-		m_lastMotor = IntakeLastMotor.FRONT;
+	public void requestFrontIntakeOffBackIntakeIn() {
+		if(intakesAreEnabled()) {
+			m_intakeFrontMotor.set(0);
+			m_intakeBackMotor.set(IntakeConstants.MAX_INTAKE_SPEED);
+			m_lastMotor = IntakeLastMotor.BACK;
+			RobotState.m_intakeDirection = IntakeDirection.BACK;
+		} else {
+			requestIntakeOff();
+		}
+	}
+
+	public void requestFrontIntakeOffBackIntakeOut() {
+		if(intakesAreEnabled()) {
+			m_intakeFrontMotor.set(0);
+			m_intakeBackMotor.set(-IntakeConstants.MAX_INTAKE_SPEED);
+			m_lastMotor = IntakeLastMotor.BACK;
+			RobotState.m_intakeDirection = IntakeDirection.BACK;
+		} else {
+			requestIntakeOff();
+		}
+	}
+
+	public void requestFrontIntakeIn() {
+		if(intakesAreEnabled()) {
+			m_intakeFrontMotor.set(IntakeConstants.MAX_INTAKE_SPEED);
+			m_lastMotor = IntakeLastMotor.FRONT;
+			RobotState.m_intakeDirection = IntakeDirection.FRONT;
+		} else {
+			requestIntakeOff();
+		}
+	}
+
+	public void requestFrontIntakeOut() {
+		if(intakesAreEnabled()) {
+			m_intakeFrontMotor.set(-IntakeConstants.MAX_INTAKE_SPEED);
+			m_lastMotor = IntakeLastMotor.FRONT;
+			RobotState.m_intakeDirection = IntakeDirection.FRONT;
+		} else {
+			requestIntakeOff();
+		}
+	}
+
+	public void requestFrontIntakeInBackIntakeOff() {
+		if(intakesAreEnabled()) {
+			m_intakeFrontMotor.set(IntakeConstants.MAX_INTAKE_SPEED);
+			m_intakeBackMotor.set(0);
+			m_lastMotor = IntakeLastMotor.FRONT;
+		} else {
+			requestIntakeOff();
+		}
+	}
+
+	public void requestFrontIntakeOutBackIntakeOff() {
+		if(intakesAreEnabled()) {
+			m_intakeFrontMotor.set(IntakeConstants.MAX_INTAKE_SPEED);
+			m_intakeBackMotor.set(0);
+			m_lastMotor = IntakeLastMotor.FRONT;
+		} else {
+			requestIntakeOff();
+		}
 	}
 
 	public IntakeLastMotor getLastMotor() {
 		return m_lastMotor;
 	}
 
-	public void intakeOff() {
+	public void requestIntakeOff() {
 		m_intakeMotorGroup.set(0);
 		RobotState.m_intakeDirection = IntakeDirection.NONE;
 	}
 
-	public void intakeIn() {
-		m_intakeMotorGroup.set(1);
-		m_lastMotor = IntakeLastMotor.BOTH;
-		RobotState.m_intakeDirection = IntakeDirection.BOTH;
+	public void requestIntakeIn() {
+		if(intakesAreEnabled()) {
+			m_intakeMotorGroup.set(1);

```

Comment:
Would be nice to use a constant and not a magic number.

Reply:
Agreed. A lot of `IntakeOnOffSubsystem` could be refactored in my opinion. In the interest of student involvement and education I attempted to keep codebase changes to a minimum. Stylistic choices like this are definitely something we should cover with the entire team in the future.